src/EditorFeatures/Core/DocumentationComments/AbstractDocumentationCommentCommandHandler.cs

```diff
@@ -117,19 +117,48 @@ public CommandState GetCommandState(TypeCharCommandArgs args, Func<CommandState>
 
     public void ExecuteCommand(TypeCharCommandArgs args, Action nextHandler, CommandExecutionContext context)
     {
-        // Ensure the character is actually typed in the editor
-        nextHandler();
+        // Don't execute in cloud environment, as we let LSP handle that
+        if (args.SubjectBuffer.IsInLspEditorContext())
```

Comment:
The trigger check below looks pretty cheap. If this check is more expensive, we may want to move it after the trigger check.